src/main/java/io/sniffy/Spy.java

```diff
@@ -55,51 +82,97 @@ protected void resetExecutedStatements() {
             case OTHERS:
                 statements = new ArrayList<StatementMetaData>();
                 for (StatementMetaData statement : executedStatements) {
-                    if (statement.owner == this.owner) {
+                    if (statement.owner != this.owner) {
                         statements.add(statement);
                     }
                 }
                 break;
             case ANY:
-            default:
                 statements = new ArrayList<StatementMetaData>(executedStatements);
                 break;
+            default:
+                throw new IllegalArgumentException(String.format("Unknown thread matcher %s", threadMatcher.getClass().getName()));
         }
         return Collections.unmodifiableList(statements);
     }
 
-    Spy() {
+    Spy(boolean spyCurrentThreadOnly) {
         owner = Thread.currentThread();
         selfReference = Sniffer.registerSpy(this);
+        this.spyCurrentThreadOnly = spyCurrentThreadOnly;
         reset();
     }
 
-    private void initNumberOfQueries() {
-        this.initialCount = new Counter(Sniffer.COUNTER);
-        this.initialThreadLocalCount = new Counter(Sniffer.THREAD_LOCAL_COUNTER.get());
-    }
-
-    private List<Expectation> expectations = new ArrayList<Expectation>();
-
     /**
      * Wrapper for {@link Sniffer#spy()} method; useful for chaining
      * @return a new {@link Spy} instance
      * @since 2.0
      */
     public Spy reset() {
         checkOpened();
-        initNumberOfQueries();
         resetExecutedStatements();
+        resetSocketOpertions();
         expectations.clear();
         return self();
     }
 
+    public Map<SocketMetaData, SocketStats> getSocketOperations(Threads threadMatcher) {
+        return getSocketOperations(threadMatcher, null, true);
+    }
+
+    public Map<SocketMetaData, SocketStats> getSocketOperations(Threads threadMatcher, String address, boolean removeStackTraces) {
+
+        String hostName = null;
+        Integer port = null;
+
+        if (null != address) {
+            if (-1 != address.indexOf(':')) {
+                String[] split = address.split(":");
+                hostName = split[0];
+                port = Integer.valueOf(split[1]);
+            } else {
+                hostName = address;
+            }
+        }
+
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [An empty statement (semicolon) not part of a loop](https://www.codacy.com/app/sniffy/sniffy/file/2507982246/issues/source?bid=3005960&fileBranchId=3034483#l138)